Realm/Realm/Dynamic/MetaRealmObject.cs

```diff
@@ -127,15 +100,13 @@ public override DynamicMetaObject BindGetMember(GetMemberBinder binder)
                         getter = GetGetMethod(DummyHandle.GetList<byte[]>);
                         break;
                     case PropertyType.Date:
-                        if (property.Type.IsNullable())
-                        {
-                            getter = GetGetMethod(DummyHandle.GetList<DateTimeOffset?>);
-                        }
-                        else
-                        {
-                            getter = GetGetMethod(DummyHandle.GetList<DateTimeOffset>);
-                        }
-
+                        getter = property.Type.IsNullable() ? GetGetMethod(DummyHandle.GetList<DateTimeOffset?>) : GetGetMethod(DummyHandle.GetList<DateTimeOffset>);
+                        break;
+                    case PropertyType.ObjectId:
+                        getter = property.Type.IsNullable() ? GetGetMethod(DummyHandle.GetList<ObjectId?>) : GetGetMethod(DummyHandle.GetList<ObjectId>);
+                        break;
+                    case PropertyType.Decimal:
+                        getter = property.Type.IsNullable() ? GetGetMethod(DummyHandle.GetList<Decimal128?>) : GetGetMethod(DummyHandle.GetList<Decimal128>);
```

Comment:
These were forgotten (and apparently untested 🙈 @papafe you should add a case for UUID once this goes in.